pkg/controllers/v1alpha1/mpi_job_controller_test.go

```diff
@@ -549,8 +551,10 @@ func TestLauncherFailed(t *testing.T) {
 	mpiJob := newMPIJob("test", int32Ptr(64), &startTime, nil)
 	f.setUpMPIJob(mpiJob)
 
+	mpiJob.Status.LauncherStatus = kubeflow.LauncherFailed
 	launcher := newLauncher(mpiJob, "kubectl-delivery")
 	launcher.Status.Failed = 1
+	launcher.Status.Conditions = []batchv1.JobCondition{batchv1.JobCondition{Type: batchv1.JobComplete, Status: corev1.ConditionTrue}}
```

Comment:
Condition should be `Failed` here?

Reply:
if i set Failed, it will not pass ci test. Prev logic is `done = launcher != nil && mpiJob.Status.Failed = 1 || mpiJob.Status.Successed = 1`, prev set `launcher.Status.Failed = 1`. When the launcher is not null, done will be true. So i just to make sure the done is true.

Reply:
A little confuse what's the difference of Failed and Completed here? the new isJobFinished func we cover all this condition

Reply:
`job.Status.Failed`  is the num of failed pod,  it may get nums < backoffLimit. When launcher is not null and job.Status.Failed = 1 , `done` will be true. But in real world, it will restart laucher because the `backofflimit=6 `default .

Reply:
Yes, I know this case, and in v1alpha2 I also have change to isJobFinished. Here my question is the test func is `TestLauncherFailed`, but we set the condition in L557 to JobComplete which mean success, not the Failed?

Reply:
Yep, you are right. I should set it be Failed and delete `launcher.Status.Failed = 1`. :)

Reply:
Yes, this make sense to me :)  If you are interesting with this, suggest you to try v1alpha2, the api is more clear :)

Reply:
You can change the v1alphav2 test file. @fisherxu

Reply:
Yeah sure.